src/meta/src/hummock/manager/mod.rs

```diff
@@ -2341,80 +2356,65 @@ where
+                            *throughput / checkpoint_secs
+                                > self.env.opts.table_write_throughput_threshold
+                        });
+                        is_low_write_throughput = history.iter().any(|throughput| {
+                            *throughput / checkpoint_secs
+                                < self.env.opts.min_table_split_write_throughput
                         });
-                        is_low_write_throughput = window_total_size
-                            < (HISTORY_TABLE_INFO_WINDOW_SIZE as u64)
-                                * self.env.opts.min_table_split_write_throughput;
                     }
                 }
                 let state_table_size = *table_size;
 
+                if is_low_write_throughput {
+                    continue;
+                }
+
                 if state_table_size < self.env.opts.min_table_split_size
                     && !is_high_write_throughput
                 {
                     continue;
                 }
 
                 let parent_group_id = group.group_id;
-                let mut target_compact_group_id = None;
-                let mut allow_split_by_table = false;
-                if state_table_size < self.env.opts.split_group_size_limit
-                    && is_low_write_throughput
-                {
-                    // do not split a large table and a small table because it would increase IOPS
-                    // of small table.
-                    if parent_group_id != default_group_id && parent_group_id != mv_group_id {
-                        let rest_group_size = group.group_size - state_table_size;
-                        if rest_group_size < state_table_size
-                            && rest_group_size < self.env.opts.min_table_split_size
-                        {
-                            continue;
-                        }
-                    } else {
-                        for group in &group_infos {
-                            // do not move to mv group or state group
-                            if !group.split_by_table || group.group_id == mv_group_id
-                                || group.group_id == default_group_id
-                                || group.group_id == parent_group_id
-                                // do not move state-table to a large group.
-                                || group.group_size + state_table_size > group_size_limit
-                                // do not move state-table from group A to group B if this operation would make group B becomes larger than A.
-                                || group.group_size + state_table_size > group.group_size - state_table_size
-                            {
-                                continue;
-                            }
-                            target_compact_group_id = Some(group.group_id);
-                        }
-                        allow_split_by_table = true;
```

Comment:
What was `allow_split_by_table` used prior to this PR? Is it only used to control whether we need to do `config.max_bytes_for_level_base *= 4;`? If yes, it seems that we can remove this parameter in `move_state_table_to_compaction_group` completely.

Reply:
If we do not hope move several state-table to one group, `allow_split_by_table ` will be never used anymore

Reply:
allow_split_by_table doesn't make sense after this pr, we make sure that there is only one table per new cg.